lib/dash/content_protection.js

```diff
@@ -320,6 +320,50 @@ shaka.dash.ContentProtection = class {
     return ContentProtection.getLaurl_(rootElement);
   }
 
+
+  /**
+   * Gets a PlayReady initData from a content protection element
+   * containing a PlayReady Pro Object
+   *
+   * @param {shaka.dash.ContentProtection.Element} element
+   * @return {?Array.<shaka.extern.InitDataOverride>}
+   */
+  static getInitDataFromPro(element) {
+    const proNode = shaka.util.XmlUtils.findChildNS(
+        element.node, 'urn:microsoft:playready', 'pro');
+    if (!proNode) {
+      return null;
+    }
+    const fromBase64 = shaka.util.Uint8ArrayUtils.fromBase64;
+    const type = fromBase64(btoa('pssh'));
```

Comment:
Clever, but I think it would be better to just hard-code the bytes to make it clear what this is doing.  You could also use a 32-bit number:

```js
psshData.setUint32(byteCursor, 0x70737368);  // 'pssh'
```